api/envoy/service/auth/v3/external_auth.proto

```diff
@@ -60,8 +60,20 @@ message DeniedHttpResponse {
   string body = 3;
 }
 
+// TODO: Should this be in the core API?
```

Comment:
This is an open question. I feel like it could be valuable alongside HeaderValueOption.

Reply:
Yes, that would be a good location, but I think the `remove` semantics are a bit weird. Would it be cleaner to structure this similar to headers, with `query_parameters_to_add`, `query_parameter_to_remove`?

Reply:
I think I agree with @htuch, though with introducing a new structure the same as `HeaderValue`? But with the `QueryParameter` (?) as its name. We can do that via what is suggested (adding `query_parameters_to_add`, `query_parameter_to_remove`) by Harvey.

Reply:
Will do